pkg/controllers/netpol/network_policy_controller.go

```diff
@@ -144,24 +142,26 @@ func (npc *NetworkPolicyController) Run(healthChan chan<- *healthcheck.Controlle
 	glog.Info("Starting network policy controller")
 	npc.healthChan = healthChan
 
-	// loop forever till notified to stop on stopCh
-	for {
-		select {
-		case <-stopCh:
-			glog.Info("Shutting down network policies controller")
-			return
-		default:
+	// Full syncs of the network policy controller take a lot of time and can only be processed one at a time,
+	// therefore, we start it in it's own goroutine and request a sync through a single item channel
+	glog.Info("Starting network policy controller full sync goroutine")
+	go func(fullSyncRequest <-chan struct{}, stopCh <-chan struct{}) {
+		for {
+			select {
```

Comment:
go's select case statement does not have priorty. So its better to give priority to `case <-stopCh` else we run into corner case where full sync's can continue to run inspite of stopCh is closed.